packages/monaco/src/vim-keybindings.ts

```diff
@@ -0,0 +1,38 @@
+import { type VimMode } from 'monaco-vim';
+
+export default function registerKeyBindings(vim: typeof VimMode) {
+  vim.Vim.defineAction('hover', function (ctx) {
+    ctx.editor.trigger('action', 'editor.action.showHover', null);
+  });
+  vim.Vim.mapCommand('K', 'action', 'hover');
+
+  vim.Vim.defineAction('goToDeclaration', function (ctx) {
+    ctx.editor.trigger('action', 'editor.action.goToDeclaration', null);
+  });
+  vim.Vim.mapCommand('gd', 'action', 'goToDeclaration');
+
+  vim.Vim.defineAction('goToReferences', function (ctx) {
+    ctx.editor.trigger('action', 'editor.action.goToReferences', null);
+  });
+  vim.Vim.mapCommand('gr', 'action', 'goToReferences');
+
+  vim.Vim.defineAction('goToImplementation', function (ctx) {
+    ctx.editor.trigger('action', 'editor.action.goToImplementation', null);
+  });
+  vim.Vim.mapCommand('gi', 'action', 'goToImplementation');
+
+  vim.Vim.defineAction('goToLink', function (ctx) {
+    ctx.editor.trigger('action', 'editor.action.openLink', null);
+  });
+  vim.Vim.mapCommand('gx', 'action', 'goToLink');
+
+  vim.Vim.defineAction('nextDiagnostic', function (ctx) {
+    ctx.editor.trigger('action', 'editor.action.marker.next', null);
+  });
+  vim.Vim.mapCommand(']d', 'action', 'nextDiagnostic');
+
+  vim.Vim.defineAction('prevDiagnostic', function (ctx) {
+    ctx.editor.trigger('action', 'editor.action.marker.prev', null);
+  });
+  vim.Vim.mapCommand('[d', 'action', 'prevDiagnostic');
```

Comment:
We can probably avoid duplicating the same things over and over by adding a utility function and looping over a list keybinds. Something like:

```suggestion
  type Keybind = {
    keybind: string
    name: string
    action: string
  }

  function registerKeyBinding({ keybind, name, action }: Keybind) {
    vim.Vim.defineAction(name, ctx => {
      ctx.editor.trigger('action', action, null);
    });
    vim.Vim.mapCommand(keybind, 'action', name);
  }
  
  const KEYBINDS: Keybind[] = [...];
  KEYBINDS.forEach(registerKeyBinding);
```